File: kotlin/src/main/kotlin/com/joshy/hywrapper/util/UuidUtils.kt
package com.joshy.hywrapper.util

object UuidUtils {
    fun undash(uuid: String): String {
        return uuid.replace("-", "").lowercase()
    }

    fun dash(uuid: String): String {
        val undashed = undash(uuid)
        if (undashed.length != 32) return uuid

        return buildString {
            append(undashed.substring(0, 8))
            append("-")
            append(undashed.substring(8, 12))
            append("-")
            append(undashed.substring(12, 16))
            append("-")
            append(undashed.substring(16, 20))
            append("-")
            append(undashed.substring(20, 32))
        }
    }

    fun isValid(uuid: String): Boolean {
        val undashed = undash(uuid)
        return undashed.length == 32 && undashed.all { it in '0'..'9' || it in 'a'..'f' }
    }
}

File: typescript/src/utils/uuid_utils.ts
export class UuidUtils {
    static undash(uuid: string): string {
        return uuid.replace(/-/g, "").toLowerCase();
    }

    static dash(uuid: string): string {
        const undashed = this.undash(uuid);
        if (undashed.length !== 32) return uuid;

        return (
            undashed.substring(0, 8) +
            "-" +
            undashed.substring(8, 12) +
            "-" +
            undashed.substring(12, 16) +
            "-" +
            undashed.substring(16, 20) +
            "-" +
            undashed.substring(20, 32)
        );
    }

    static isValid(uuid: string): boolean {
        const undashed = this.undash(uuid);
        return undashed.length === 32 && /^[0-9a-f]+$/.test(undashed);
    }
}
